### DISCUSSION TITLE
Workflow: Register Archaeologist

### DISCUSSION THREAD

**Discussion by adamgall** [sarcophagus-org/sarcophagus-v2-specs#14] [Ideas]
This post is to think through and describe what it means for an archaeologist to be "registered" into the system.

In V2, archaeologists are a lot more "real time" and act like a true p2p network with embalmers. Therefore, less archaeologist state will need to be kept on-chain than in V1.

An archaeologist can be considered "registered" as long as:

1. They have posted some bond into the system, and have bond available to take new sarcophagi.
2. They are responsive within the p2p network, and perform the necessary communications with embalmers to get through the "create sarcophagus" workflow.

**Comment by adamgall** [comment]
In previous discussions (not on GitHub), we've talked about the idea of an archaeologist uploading their xPub key to the chain, so that embalmers can calculate their next public key for any given new sarcophagus.

But, what if the embalmer just grabbed that "next key" directly from the archaeologist during the "new sarcophagus" workflow?

Is there a downside to this method?  It does require that the embalmer communicates with an archaeologist directly, before creating the initial "create sarcophagus" transaction, but in our new p2p model that's probably ok.

**Comment by adamgall** [comment]
One thing I like about this model, is that if an archaeologist wants to not accept new sarcophagi for some period of time, all they need to do is turn off their service.

Technically, from the archaeologist's point of view, this also works in V1. But, in the V1 design, it would result in embalmers wasting gas from new "create sarcophagus" transactions with an archaeologist that was offline and would never be able to confirm the new sarcophagus.

In this new design, since public keys need to come directly from the archaeologist service, and archaeologist public keys are needed to even make the first "create sarcophagus" transaction, it results in a safer and more economical experience for all actors.

**Reply by sethhrbek** [reply]
I agree with this approach (direct communication for the "next" pub key). In v1, we had added a "liveness" https endpoint for the arch service which would indicate whether any given arch was running. This key communication could act as the "liveness" check in v2.
